Assets/Scripts/Towers/LightningShootable.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class LightningShootable : MonoBehaviour
+{
+
+    //*********************************************************************************************************
+    //This script should be called by lightning tower using the shoot() function from the animation controller.
+    //*********************************************************************************************************
+
+
+    //The radius for the turret and all it's targets.
+    public float lightningRadius = 4;
+    //Damage per zap.
+    public float damage = 10;
+    //Lightning VFX
+    public GameObject drainRay;
+    //LighntingEffect duration
+    public float effectDuration = 0.3f;
+    //What layers the collider will check in, should be player and enemy layers.
+    public LayerMask shockLayers;
+
+
+    //Marked objects
+    private List<GameObject> zappTargets = new List<GameObject>();
+
+
+    public void Shoot()
+    {
+        CheckZap(this.gameObject);
+
+        foreach (var zap in zappTargets)
+        {
+            zap.GetComponent<HealthLogic>().DealDamage(damage);
+        }
+        //Clear all marked objects.
+        zappTargets.Clear();
+    }
+
+
+    //Marks soon-to-be zapped objects and adds a VFX.
+    private void AddZap(GameObject target)
+    {
+        if (target.GetComponent<HealthLogic>() && !zappTargets.Contains(target))
+        {
+            GameObject previousTarget = null;
+            if (zappTargets.Count > 0)
+            {
+                previousTarget = zappTargets[zappTargets.Count - 1];
+            }
+            else
+            {
+                previousTarget = this.gameObject;
+            }
```

Comment:
This can be shortened to the following:
```suggestion
            GameObject previousTarget = zappTargets.LastOrDefault() ?? gameObject;
```
This would require adding `using System.Linq;` to the top of the file, though 🙂

Reply:
Didn't know this, very cool!